python/python/lance/dataset.py

```diff
@@ -512,12 +513,21 @@ def __init__(self, ds: LanceDataset):
         self._columns = None
         self._nearest = None
         self._batch_readahead = None
+        self._ordered_scan = True
 
     def batch_readahead(self, nbatches: Optional[int] = None) -> ScannerBuilder:
         if nbatches is not None and int(nbatches) < 0:
             raise ValueError("batch_readahead must be non-negative")
         self._batch_readahead = nbatches
         return self
+    
+    def ordered_scan(self, ordered_scan: bool = True) -> ScannerBuilder:
+        """
+        Whether to scan the dataset in order. If set to False, the scanner may
+        read fragments concurrently and yield batches out of order.
```

Comment:
could you also mention that "out of order" scan might yield to better performance , so that users know what the benefit of this flag is?